lib/Doctrine/DBAL/Schema/Index.php

```diff
@@ -251,11 +247,7 @@ public function overrules(Index $other)
             return false;
         }
 
-        if ($this->spansColumns($other->getColumns()) && ($this->isPrimary() || $this->isUnique()) && $this->samePartialIndex($other)) {
-            return true;
-        }
-
-        return false;
+        return $this->spansColumns($other->getColumns()) && ($this->isPrimary() || $this->isUnique()) && $this->samePartialIndex($other);
```

Comment:
Same as above, this condition could be split into a series for better readability.